### PR TITLE
App tests

### PR DESCRIPTION
**Pull request by Wmaileq**
## Description of the change

- App.js tests
- Coverage increased from 26.52% to 35.669%

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

> Part of #25 

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing against realistic data has been performed locally

### Code review

These boxes should be checked by reviewers prior to merging:

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [x] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: src/App.test.js [hunk 1]
```diff
@@ -15,18 +15,106 @@
...
+      "/community/revocations"
+    );
+    const user = {
+      "https://dashboard.recidiviz.org/app_metadata": {
+        state_code: MO,
+      },
+    };
+    useAuth0.mockReturnValue({
+      user,
+      isAuthenticated: true,
```
Comment (jovergaag) [inline src/App.test.js:62]: Can you add a test case for when `isAuthenticated` is false and when `loading` is true. We just want to be sure that nothing is rendered before the user is authenticated. We could also probably want a test to ensure that the user is redirected to the login page.

If this is too difficult to write tests for in demo mode, let me know and I can take over those last couple.

**Review by jovergaag — CHANGES_REQUESTED** [review]
Just a couple of non-happy-path tests that we should look into.

Thanks!

File: src/App.test.js (hunk 1 above)
Comment (Wmaileq) [reply src/App.test.js:72]: Tests added!
Thinking about what we need here (integrational or unit tests) came to the idea that unit tests here would just cover internal react-router logic, what we don't really need.
So I've covered `App` in a bundle with the `TenantRoutes` and the `StateCodeProvider` components that resulted in having almost 100% coverage on the `TenantRoutes`.
Despite this, I think we need to do unit tests for `TenantRoutes` and `StateCodeProvider` anyways.

**Review by jovergaag — CHANGES_REQUESTED** [review]
Just getting clarification on `TenantRoutes` test.

File: src/components/TenantRoutes.js [hunk 2]
```diff
@@ -56,21 +56,15 @@ const TenantRoutes = ({ children }) => {
     return null;
   }
 
-  let element = null;
-
-  Children.forEach(children, (child) => {
+  return Children.toArray(children).reduce((node, child) => {
```
[the remark this answers is not in the document]
Comment (Wmaileq) [reply src/components/TenantRoutes.js:59]: @jovergaag 
image
TenantRoutes completely covered by this PR :)

[the remark this answers is not in the document]
Comment (jovergaag) [reply src/components/TenantRoutes.js:59]: Oh! I haven't actually logged into Coveralls, or worked with it before. I'm a little surprised that `TenantRoutes` is considered covered. 

Let's go ahead and merge this and you can continue adding tests for the uncovered files, then we can go back and add unit tests for `TenantRoutes` later. We will be continuing to refactor some of the other components so it feels more important to have them covered as we refactor. 

Eventually I would like to change the routes to add `:state_code` to the route and maybe even the filter state - at that point the `TenantRoute` tests will be more important but that is much further down the road. I actually don't think `TenantRoutes` is a great implementation and would like to find a different solution overall...

File: __helpers__/mockWithTestId.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import React from "react";
```
Comment (jovergaag) [inline __helpers__/mockWithTestId.js:1]: Great idea.

**Review by jovergaag — APPROVED** [review]
Thank you!!

**Pull request merged**